packages/data-layer/src/data.types.ts

```diff
@@ -51,6 +67,115 @@ export type Project = {
   applicationIndex: number;
 };
 
+/**
+ * The project role type for v2
+ *
+ * @remarks
+ *
+ * This is the type for v2 project roles which get created on Allo as `Profiles`.
+ *
+ * @example
+ *
+ * ```ts
+ * const projectRole: ProjectRole = {
+ *  project: {
+ *   chainId: 1,
+ *   createdAtBlock: 1,
+ *   registryAddress: "0x123",
+ *   projectNumber: 1,
+ *   tags: ["allo-v2"],
+ *  },
+ *   projectId: "0x123",
+ * };
+ *
+ * ```
+ */
+export type ProjectRole = {
+  project: {
+    chainId: number;
+    createdAtBlock: number;
+    registryAddress: string;
+    projectNumber: number;
+    tags: string[];
+  };
+  projectId: string;
+};
+
+/**
+ * The project type for v2
+ *
+ * @remarks
+ *
+ * This is the type for v2 projects which get created on Allo as `Profiles`.
+ *
+ */
+export type v2Project = {
```

Comment:
If we're using the indexer a V1 project and a v2 project should be the same, could we call this just Project? or is there another reason why it's called v2

Reply:
The Project object in the types is use more for the project application in this context - open to how we can name them.